### ISSUE TITLE
datetime_string cannot handle floating point seconds

### ISSUE THREAD

**Issue by ufnalbartosz** [fnogatz/date_time#4]
Hey, I think the following snippet has a bug.

```prolog
?- datetime_get(now, D), datetime_string(D, F, S).
false.
```

It can be reproduced this way:
```prolog
?- datetime_get(now, D).
D = datetime(2022, 3, 3, 18, 55, 57.44183659553528).
```

This snippet returns `false`:
```prolog
?- datetime_string(datetime(2022, 3, 3, 18, 55, 57.44183659553528), F, S).
false.
```

However, after I removed the floating-point seconds:
```prolog
?- datetime_string(datetime(2022, 3, 3, 18, 55, 57), F, S).
F = 'y/m/d',
S = "2022/3/3 18:55:57".
```
The formatted string works as expected.

**Comment by fnogatz** [comment]
Thank you for reporting this, I'll have a look at it. To be sure, what result would you expect from your goal `datetime_string(datetime(2022, 3, 3, 18, 55, 57.44183659553528), F, S).`?

**Comment by ufnalbartosz** [comment]
I've used the seconds. Here is my solution. However, any precision would be admirable.

```prolog
date_time:datetime_get(now, datetime(Y, M, D, Hr, Mn, X)), 
    Sc is integer(X),
    date_time:datetime_string(datetime(Y, M, D, Hr, Mn, Sc), 'y/m/d', DateTimeNow).
```

**Comment by fnogatz** [comment]
Fixed via 8bcc8505257988e6358bc00b94f9a466712b48b7 in #5.
